R/utils.R

```diff
@@ -324,6 +324,145 @@ date_to_numeric_modulus <- function(dt, date_column, timestep) {
   return(dt[])
 }
 
+#' Set caching location for Stan models
+#'
+#' This function allows the user to set a cache location for
+#' Stan models rather than a temp directory. This can reduce the
+#' need for model compilation on every new model run across sessions.
+#'
+#' @param path A valid filepath representing the desired cache location
+#'
+#' @return The string of the filepath set
+#' @examples
+#' # Set to local directory
+#' dir.create("test")
+#' enw_set_cache("test")
+#' enw_get_cache()
+#' # Use  the package cache in R >= 4.0
+#' if(R.version.string >= 4.0) {
+#'   enw_set_cache(tools::R_user_dir(package = "epinowcast", "cache"))
+#' }
+#' @importFrom cli cli_abort cli_alert
+#' @export
+
+enw_set_cache <- function(path = NULL) {
+
+  if (is.null(path)) {
+    cli::cli_abort("`path` must be a valid file path.")
+  }
+
+  prior_cache <- Sys.getenv("enw_cache_location", unset = "", names = NA)
+
+  if (!check_environment_setting(prior_cache)) {
+    cli::cli_alert("{prior_cache} exists and will be overwritten")
+  }
+  env_contents_active <- enw_get_environment_contents()
+
+  #!TODO should we normalise the path?
```

Comment:
I don't actually have a clue what this means. Explain to me like I'm 5 please!

Reply:
Normalisation will expand the path to the fully specified version and will input the platform specific escape patterns (double back slashes on windows) e.g., `normalizePath("..")` -> `C:\\Users\\medewitt\\project\\epinowcast`. I think normalising will result in a safer experience given that Windows can do some wonky things without it. The downside is might be for multi-user systems that run on a generic Renviron file so the cache location is actually different for different users (maybe) or if a user always wants to have a relative path for the cache (e.g., a model directory within their project where the stan code that was used is local to a project). In the latter case they would need to set the cache for each new project or use a local Renviron at the level of the project. Might be overthinking it.